### PR TITLE
[OTE-135] add roundtable task for CLOSE_ONLY -> BLOCKED compliance status transition

### PR DESCRIPTION
**Pull request by dydxwill**
### Changelist
add roundtable task for CLOSE_ONLY -> BLOCKED compliance status transition

### Test Plan
unit tested

### Author/Reviewer Checklist
- [ ] If this PR has changes that result in a different app state given the same prior state and transaction list, manually add the `state-breaking` label.
- [ ] If the PR has breaking postgres changes to the indexer add the `indexer-postgres-breaking` label.
- [ ] If this PR isn't state-breaking but has changes that modify behavior in `PrepareProposal` or `ProcessProposal`, manually add the label `proposal-breaking`.
- [ ] If this PR is one of many that implement a specific feature, manually label them all `feature:[feature-name]`.
- [ ] If you wish to for mergify-bot to automatically create a PR to backport your change to a release branch, manually add the label `backport/[branch-name]`.
- [ ] Manually add any of the following labels: `refactor`, `chore`, `bug`.

### CODE REVIEW

File: indexer/services/roundtable/src/tasks/perform-compliance-status-transitions.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+  // Update addresses status to BLOCKED
+  const updatedAddresses: ComplianceStatusFromDatabase[] = await Promise.all(
+    addressesToUpdate.map((address) => ComplianceStatusTable.update({
+      address,
+      status: ComplianceStatus.BLOCKED,
+      updatedAt: new Date().toISOString(),
+    }),
+    ),
+  ) as ComplianceStatusFromDatabase[];
```
Comment (vincentwschau) [inline indexer/services/roundtable/src/tasks/perform-compliance-status-transitions.ts:47]: nit: Non-blocking, but `bulkUpsert` probably would be a bit more efficient to save on round-trips to the DB if updating multiple addresses.

File: indexer/services/roundtable/__tests__/tasks/perform-compliance-status-transitions.test.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        address: testConstants.defaultAddress,
+        status: ComplianceStatus.CLOSE_ONLY,
+        createdAt: DateTime.utc().toISO(),
+        updatedAt: DateTime.utc().toISO(),
+      }),
+    ]);
+
+    await performComplianceStatusTransitionsTask();
+
+    // Assert the status was updated to BLOCKED
```
Comment (vincentwschau) [inline indexer/services/roundtable/__tests__/tasks/perform-compliance-status-transitions.test.ts:72]: nit: Assert that `updatedAt` was updated correctly, and also that the `defaultAddress` wasn't updated.

File: indexer/services/roundtable/src/tasks/perform-compliance-status-transitions.ts (hunk 1 above)
Comment (dydxwill) [reply indexer/services/roundtable/src/tasks/perform-compliance-status-transitions.ts:47]: Will address this in a follow-up PR, I think `bulkUpsert` function needs some changes.

File: indexer/services/roundtable/src/tasks/perform-compliance-status-transitions.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+import {
+  ComplianceStatusFromDatabase,
+  ComplianceStatusTable,
+  ComplianceStatus,
+} from '@dydxprotocol-indexer/postgres';
+
+import config from '../config';
+
+// eslint-disable-next-line max-len
+const CLOSE_ONLY_TO_BLOCKED_DAYS_IN_MS: number = config.CLOSE_ONLY_TO_BLOCKED_DAYS * 24 * 60 * 60 * 1000;
```
[the remark this answers is not in the document]
Comment (dydxwill) [reply indexer/services/roundtable/src/tasks/perform-compliance-status-transitions.ts:13]: Done

File: indexer/services/roundtable/__tests__/tasks/perform-compliance-status-transitions.test.ts (hunk 2 above)
Comment (dydxwill) [reply indexer/services/roundtable/__tests__/tasks/perform-compliance-status-transitions.test.ts:73]: done

**Pull request merged**